blog/_posts/2024-08-21-scala-3.5.0-released.md

```diff
@@ -0,0 +1,159 @@
+
+### Support for pipelined builds
+
+Scala 3.5.0 supports pipelined compilation. It can be enabled by setting `ThisBuild/usePipelining := true` in sbt build definition. This can result in significant speedups in compilation time for multi-module projects.
+
+You can learn more about how the pipelined compilation works and what benefits you can expect from [the talk by Jamie Thompson](https://www.youtube.com/watch?v=1uuFxEAiPuQ&t=1473s).
+
+### `var` in refinements
+
+Until now, only types, `val`s, and `def`s were allowed in type refinements. In 3.5.0, `var`s can also be declared.
+
+```scala
+type A = { var number: Int }
+```
+
+is now legal and equivalent to
+
+```scala
+type A = { def number: Int, def number_=($number: Int): Unit }
+```
+
+This change can simplify libraries based on metaprogramming using type refinements.
+
+### Support for binary integer literals
+
+Integer literals can now start with `0b` or `0B`. They will be interpreted as numbers in base 2.
+
+```scala
+assert(0B1 == 1)
+assert(0B10 == 2)
+assert(0B_1000_0010 == 130)
+assert(0B_1000_0010 == 0x82)
+```
+
+### Experimental: named tuples
+
+Scala 3.5 introduces experimental support for named tuples. The feature hidden behind the `language.experimental.namedTuples` import, allows you to give meaningful names to tuple elements and use those nemes during constructing, destructuring, and pattern matching.
+
+```scala
+import scala.language.experimental.namedTuples
+
+type Point = (x: Int, y: Int)
+val point: Point = (x = 1, y = 2)
+val is_it_point = (x = 5, y = 3)
+val it_is: Point = is_it_point
+
+println(point.x) // prints 1
+
+point match
+  case (x = real, y = 0) => println(s"Real number: $real")
+  case _ => println("Point doesn't represent a real number")
+```
+
+This is an implementation of [SIP-58](https://github.com/scala/improvement-proposals/blob/d649f6e6f333cd9232d85a12bd0445d18a673f10/content/named-tuples.md).
+
+## Work on a better scheme for given prioritization
+
+Givens in Scala 3 have a peculiar problem with prioritization. The compiler tries to always select the instance with *the most specific subtype* of the requested type. This can lead to confusing situtaions, when user faces ambiguity errors in code that should intuitivel work. Changing the scheme of given priritization to always select the instance with *the most general subtype* that satisfies the context bound, would resolve such cases. We have conducted experiments that showed that the proposed scheme will result in a more intuitive and predictable given resolution. The negative impact on the existing projects is very small. We have tested 1500 open-source libraries, and new rules are causing problems for less than a dozen of them. We have already submitted PRs with changes that will make them work the same way under both the current and proposed rules.
```

Comment:
```suggestion
Givens in Scala 3 have a peculiar problem with prioritization. The compiler tries to always select the instance with *the most specific subtype* of the requested type. This can lead to confusing situations, when user faces ambiguity errors in code that should intuitively work. Changing the scheme of given priritization to always select the instance with *the most general subtype* that satisfies the context bound, would resolve such cases. We have conducted experiments that showed that the proposed scheme will result in a more intuitive and predictable given resolution. The negative impact on the existing projects is very small. We have tested 1500 open-source libraries, and new rules are causing problems for less than a dozen of them. We have already submitted PRs with changes that will make them work the same way under both the current and proposed rules.
```